server/controllers/reports_proposed/data/account_result.js

```diff
@@ -0,0 +1,239 @@
+      var sections = (currents.length > 0) ? getSections(currents) : [];
+
+      context.profitGeneral = 0, context.profitGeneralPrevious = 0;
+
+      sections.forEach(function (section){
+        section.total = 0, section.totalPrevious = 0;
+        section.refs = getReferences(section, currents);
+
+        section.refs.forEach(function (item){
+            item.net = getProfitNet(item, currents);
+            item.net_view = numeral(item.net).format(formatDollar);
+            section.total += item.net;
+
+            item.previous = getPreviousProfit(item, previous);
+            item.previous_view = numeral(item.previous).format(formatDollar);
+            section.totalPrevious += item.previous;
+        });
+
+        section.total_view = numeral(section.total).format(formatDollar);
+        section.totalPrevious_view = numeral(section.totalPrevious).format(formatDollar);
+
+        context.profitGeneral += section.total;
+        context.profitGeneralPrevious += section.totalPrevious;
+      });
+
+      context.profitGeneral = numeral(context.profitGeneral).format(formatDollar);
+      context.profitGeneralPrevious = numeral(context.profitGeneralPrevious).format(formatDollar);
+      return sections;
+    }
+
+    function exist (obj, arr, crit){
+      return arr.some(function (item){
+        return obj[crit] == item[crit];
+      });
+    }
+
+    function getSections (currents){
+      var sections = [];
+      sections.push({
+        sectionResultId : currents[0].sectionResultId,
+        sectionResultPosition : currents[0].sectionResultPosition,
+        sectionResultLabel : currents[0].sectionResultLabel,
+        sectionResultIsCharge : currents[0].sectionResultIsCharge,
+        refs : []
+      });
+
+      for(var i = 0; i <= currents.length - 1; i++){
+        if(!exist(currents[i], sections, 'sectionResultId')){
+          sections.push({
+            sectionResultId : currents[i].sectionResultId,
+            sectionResultPosition : currents[i].sectionResultPosition,
+            sectionResultLabel : currents[i].sectionResultLabel,
+            sectionResultIsCharge : currents[i].sectionResultIsCharge,
+            refs : []
+          })
+        }
+      }
+
+      return sections;
+    }
+
+
+    function getReferences (section, currents){
+      var references = [];
+
+      for(var i = 0; i <= currents.length - 1; i++){
+        if(currents[i].sectionResultId == section.sectionResultId){
+          if(!exist(currents[i], references, 'referenceId')){
+            references.push({
+              referenceId : currents[i].referenceId,
+              referenceAbbr : currents[i].referenceAbbr,
+              referencePosition : currents[i].referencePosition,
+              referenceLabel : currents[i].referenceLabel,
+              net : 0,
+              previousNet : 0
+            });
+          }
+        }
+      }
+      return references;
+    }
+
+    function getChargeNet (reference, currents){
```

Comment:
I see you use this format multiple times.  If you want to be really fancy, you can wrap this in a function that takes in a difference function.  For example:
```js
function getNetSum(referenceId, currents, fn) {
  return currents.reduce(function (sum, item) {
     return sum + (item.referenceId === referenceId ? fn(item) : 0);
  }, 0);
}

// now you can call
netCharge = getNetSum(id, currents, function (i) { return i.generalLedgerCredit - i.generalLedgerDebit; });
netProfit = getNetSum(id, currents, function (i) { return i.generalLedgerDebit - i.generalLedgerCredit; });
```